spec/support/api/helpers.rb

```diff
@@ -19,21 +19,15 @@ def init_api_spec_env
                                       :miq_groups => [@group])
         end
 
-        def api_basic_authorize(*identifiers, user: @user.userid, password: @user.password)
+        def api_basic_authorize(*identifiers, user: @user.userid, password: @user.password, extra_identifiers: nil)
+          identifiers = identifiers.flatten.compact.map(&:to_s)
+          features    = identifiers + Array(extra_identifiers)
```

Comment:
Why can't you pass the `extra_identifiers` in the array of identifiers?  I'm not sure why we need to pass them separately.

Reply:
yeah, I couldn't find a way around this...it's basically "I need these identifiers to exist in the database, but I don't want to assign them to the user's role".

This is really only needed for the 2 tests that test rbac tenant features which are dynamic features.  For example,

```ruby
        api_basic_authorize "rbac_tenant_manage_quotas_tenant_#{tenant_omega.id}", :extra_identifiers => "rbac_tenant_manage_quotas"
```

So, `rbac_tenant_manage_quotas_tenant_1` is dynamically created based on the tenants (in this case, tenant `1`), _BUT_ it's only dynamically created if `rbac_tenant_manage_quotas` feature exists in the database.  For the purpose of the test, I need the `rbac_tenant_manage_quotas` to be in the database, but I don't want the user to have that specific role otherwise they can see all quotas from all tenants.

---

Now that I wrote this 📖 , I realize this is only in 2 specific tests, so I'm considering just seeding those specific features right in that test.

Reply:
Ok, I dropped this in favor of putting it right in the tenant_quotas_spec, and in doing so, that spec got a _lot_ simpler, because it was introducing unnecessary redundancies